lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -164,8 +165,26 @@ class AwsCompileFunctions {
         delete newFunction.Properties.VpcConfig;
       }
 
+      // add the VPC permissions to the lambda execution role if a VPC config is given
+      if (newFunction.Properties.VpcConfig) {
+        // merge in the iamPolicyLambdaVPCTemplate
+        const iamPolicyLambdaVpcTemplate = this.serverless.utils.readFileSync(
+          path.join(this.serverless.config.serverlessPath,
+            'plugins',
+            'aws',
+            'deploy',
+            'compile',
+            'functions',
+            'iam-policy-lambda-vpc-template.json')
+        );
+
+        _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
+          iamPolicyLambdaVpcTemplate);
```

Comment:
@erikerikson nice catch! Thanks for bringing this up. Will consider this when continue to work on this one.